server/logging/log.go

```diff
@@ -92,15 +93,20 @@ func (l *log) Write(ctx context.Context, stepID int64, logEntry *model.LogEntry)
 	}
 
 	s.Lock()
-	s.list = append(s.list, logEntry)
+	s.list = append(s.list, entries...)
 	for sub := range s.subs {
-		go sub.handler(logEntry)
+		select {
+		case sub.receiver <- entries:
+		default:
+			logger.Info().Msgf("subscriber channel is full -- dropping logs for step %d", stepID)
```

Comment:
Shouldn't we wait for channel be empty?

Reply:
well should be fine as the data is still in DB and can be retrieved again if needed